apps/api/src/jobs/consumers/notifications/notification.consumer.ts

```diff
@@ -82,6 +82,16 @@ export abstract class NotificationConsumer {
           `Notification with ID ${id} was not sent correctly as per provider. Another attempt will be made to send the notification`,
         );
         this.logger.log('Provider response: ' + JSON.stringify(response.result));
+
+        // Check to prevent program to constantly keep checking for confirmation status
+        if (notification.retryCount >= this.maxRetryCount) {
+          notification.deliveryStatus = DeliveryStatus.FAILED;
```

Comment:
We might also be able to omit this, because when you are throwing the error, the catch block has a similar check already in place that will set the DeliveryStatus as FAILED

Reply:
Similar situation happened where initially there was repeated webhook. Then Retested it and saw that the above issue stopped repeating after letting api run idle for a few minutes, then testing it, so removed this check, as it is redundant and we have one in catch block